### ISSUE TITLE
Why is it that every time the weight of pre-training is used, the correct rate at the beginning of the model is not as good as that of the previous training?

### ISSUE THREAD

**Issue by jiayilee65** [microsoft/singleshotpose#79]
Hello. First, thanks for your work and code!
But while I was trying to replicate your results, I had a problem.
Because of the limitations of gpu memory, I modified the batch number to 8. 
Then I ran the code, trained benchvise, and looked at the acc and ADD. After training 700 epochs, I didn't reach the correct rate you gave backup weights, so I decided to train 700 epochs with the first training weight as the initial weight. However, the start of the accuracy is not as good as that of the first one, and the difference is even great.
It puzzled me that I suspected that the parameters were not fully stored, but I read the code and found no random arguments, so I'm not sure what the reason is? Do you know the reason for this?
Thank you for your help. Looking forward to your answer.

**Comment by btekin** [comment]
Hello! To train the confidence loss, we would need to first have reasonable estimates for coordinate locations. Therefore, we have a pretraining stage, where we initially set the regularization parameter for confidence to 0 and train the network for a while, only for coordinate prediction. Once the coordinate predictions are more accurate, we then train for coordinate and confidence predictions all together. This results in better confidence estimates and consequently better pose estimation accuracy. This is the reason why in practice we have a pretraining stage. 

If you would like, you could also do this within one training run, where you set the regularization parameter for confidence loss to 0 for a certain number of epochs and then continue training afterwards in the successive epochs by training all losses together with a nonzero regularization parameter for confidence loss.

**Comment by jiayilee65** [comment]
Thank you very much for your answer.
Maybe I didn't make my question clear. My question is why I continue to train with a well-trained weight, and the accuracy of the start is different from that of a trained weight.
I think it's probably the reason for learning rate. The learning rate at the beginning of re-training is high, so it is possible to deviate from the original weight too far. Do you think so?
Thank you for your help. Looking forward to your answer.

**Comment by btekin** [comment]
A high learning rate could indeed change the weights drastically and cause the deviation you observed. About the sources of randomness in the training of a PyTorch network, you could find the discussion in [this link](https://github.com/pytorch/pytorch/issues/5627) useful.

**Comment by jiayilee65** [comment]
Thank you very much. I am now trying to change the learning rate to observe the changes, and your advice is also very useful.
I wish your work well. Good luck.
